### PR TITLE
Update PlannerMd Model and Logic for Appointment

### PR DESCRIPTION
**Pull request by jiayi1129**
- Implementation of `UniqueAppointmentList` is synonymous with `UniquePersonList`, possible abstraction of `UniqueList` or a `Set`
- Main logic for Sorted Appointment List in the comments below
- Note that the Sorted list only reflects in the UI, i.e. adding to Storage will still be O(1), `SortedList` will listen to the changes and update the UI accordingly (internal implementation)
- Added natural ordering for `Appointment Class` by implementing the `Comparable` interface
- Added some tests, but pending more tests after Appointment Test Builders are added, will likely add in end of v1.3 or v1.4.

### CODE REVIEW

File: src/main/java/seedu/plannermd/model/ModelManager.java [hunk 1]
```diff
@@ -41,6 +43,10 @@ public ModelManager(ReadOnlyPlannerMd plannerMd, ReadOnlyUserPrefs userPrefs) {
         this.userPrefs = new UserPrefs(userPrefs);
         filteredPatients = new FilteredList<>(this.plannerMd.getPatientList());
         filteredDoctors = new FilteredList<>(this.plannerMd.getDoctorList());
+
+        // Wrap the FilteredList over a SortedList such that list is always sorted
+        // UniqueAppointmentList is sorted using its natural ordering as per Appointment#compareTo
+        filteredAppointments = new FilteredList<>(this.plannerMd.getAppointmentList().sorted());
```
Comment (jiayi1129) [inline src/main/java/seedu/plannermd/model/ModelManager.java:49]: Main Bulk of logic for having the Sorted Appointment List is here.
Calling `ObservableList#sorted` returns a `SortedList`, which wraps the `ObservableList` such that any changes to `ObservableList` triggers a change in `SortedList` to sort the list. We wrap the `FilteredList` over the `SortedList` to apply any filters needed after the list is sorted.

File: src/main/java/seedu/plannermd/logic/Logic.java [hunk 2]
```diff
@@ -42,9 +42,12 @@
     /** Returns an unmodifiable view of the filtered list of patients */
     ObservableList<Patient> getFilteredPatientList();
 
-    /** Returns an unmodifiable view of the filtered list of Doctors */
+    /** Returns an unmodifiable view of the filtered list of doctors */
     ObservableList<Doctor> getFilteredDoctorList();
 
+    /** Returns an unmodifiable view of the filtered list of appointments */
+    ObservableList<Doctor> getFilteredAppointmentList();
+
```
Comment (Haozeee) [inline src/main/java/seedu/plannermd/logic/Logic.java:50]: This should return a `ObservableList<Apointment>` right

File: src/main/java/seedu/plannermd/logic/LogicManager.java [hunk 3]
```diff
@@ -76,6 +76,11 @@ public ReadOnlyPlannerMd getPlannerMd() {
         return model.getFilteredDoctorList();
     }
 
+    @Override
+    public ObservableList<Doctor> getFilteredAppointmentList() {
+        return model.getFilteredDoctorList();
```
Comment (Haozeee) [inline src/main/java/seedu/plannermd/logic/LogicManager.java:81]: This should also return an `ObservableList<Appointment>` and `model.getFilteredAppointmentList`

File: src/main/java/seedu/plannermd/model/appointment/Appointment.java [hunk 4]
```diff
@@ -83,6 +84,22 @@ public boolean isSameAppointment(Appointment otherAppointment) {
...
+     * @param otherAppointment Given Appointment to compare to, not null.
+     * @return The comparator value, negative if less, positive if greater, and zero if equal.
+     */
+    @Override
+    public int compareTo(Appointment otherAppointment) {
+        requireNonNull(otherAppointment);
+
+        return date.isEqualDate(otherAppointment.date)
+                ? session.compareStartTimeTo(otherAppointment.session)
+                : date.compareTo(otherAppointment.date);
```
Comment (Haozeee) [inline src/main/java/seedu/plannermd/model/appointment/Appointment.java:100]: So if the date and start time for the 2 appointments are the same, will they be sorted randomly? Or should we then sort them by patient name/doctor name?

Comment (Haozeee) [reply src/main/java/seedu/plannermd/model/appointment/Appointment.java:100]: I think as long as the way the appointments are ordered is consistent then it's fine. So if appointments with the same date and start time appointments are ordered a certain way, after a user restarts the app, the ordering should be the same and not change. If it currently does that, then I think it's fine as it is.

**Review by Haozeee — APPROVED** [review]
LGTM

File: src/main/java/seedu/plannermd/logic/LogicManager.java [hunk 5]
```diff
@@ -30,7 +31,7 @@
     private final Storage storage;
     private final PlannerMdParser plannerMdParser;
 
-    /**
+    /**ge
```
Comment (hpkoh) [inline src/main/java/seedu/plannermd/logic/LogicManager.java:34]: typo maybe?

File: src/main/java/seedu/plannermd/model/ModelManager.java (hunk 1 above)
Comment (hpkoh) [reply src/main/java/seedu/plannermd/model/ModelManager.java:49]: It seems like uniqueAppointmentList keeps an internal sorted list that could be returned? Is that list functionally different from what is passed into the param?

**Pull request merged**